main/src/ca/uwaterloo/flix/language/phase/TreeShaker1.scala

```diff
@@ -37,12 +37,16 @@ object TreeShaker1 {
   def run(root: Root)(implicit flix: Flix): Root = flix.phase("TreeShaker1") {
     val initReach: Set[ReachableSym] = root.entryPoints.map(ReachableSym.DefnSym.apply)
 
+    val loweringTargets: Set[ReachableSym] = root.defs.filter {
+      case (_, defn) => defn.spec.ann.isLoweringTarget
+    }.keys.map(ReachableSym.DefnSym.apply).toSet
```

Comment:
You build an intermediate map only then to grab its keySet, map over that (which probably requires equal) and then call toSet? I suspect this can be done simpler. Maybe just with a plain fold?